### PR TITLE
SDL: add idiotic workaround to be less blurry on HiDPI

### PR DESCRIPTION
**Pull request by Frenzie**
Fixes https://github.com/koreader/koreader-base/issues/1300

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1602)

### CODE REVIEW

**Comment by Frenzie** [comment]
NB I had the displeasure of looking at macOS bad blurry scaling for the past half hour and I'd like to take this opportunity to state to the person I don't recall who told me that macOS is sharp that they are completely and utterly bloody wrong.

a. It sizes things up to 2x so it will *never not be blurry*.
b. If you go beyond 2x it *doesn't go to 3x so it's blurry in the **bad** way*

**Comment by Frenzie** [comment]
Mac resolutions.

The way it works is:

* 2160 x 3840 is 1x
* everything else is 2x

**Comment by Frenzie** [comment]
Anyway, on Windows things have been more or less good since Windows 8 (although they were better in 7) and there's a glimmer of light in XWayland of all places, and I understand Wayland itself will also start to behave sanely at some point in the future, so we can go back to the world of 2015 where everything "HiDPI" worked the way it should:
image

File: ffi/framebuffer_SDL2_0.lua [hunk 1]
```diff
@@ -29,6 +29,17 @@ end
 function framebuffer:resize(w, h)
     w = w or SDL.w
     h = h or SDL.h
+    local output_w = ffi.new("int[1]", 0)
+    local output_h = ffi.new("int[1]", 0)
+    if SDL.SDL.SDL_GetRendererOutputSize(SDL.renderer, output_w, output_h) == 0 and tonumber(output_w[0]) ~= w then
```
Comment (NiLuJe) [inline ffi/framebuffer_SDL2_0.lua:34]: I assume the double SDL is an unfortunate side-effect of our own ffi wrapper being named SDL, and the C namespace for the actual SDL lib being named SDL too?

**Comment by NiLuJe** [comment]
I think I grok the intent, but I'm unfamiliar with both SDL internals and HiDPI woes in general to say much about it, but I imagine it's at least less worse than before ;).

File: ffi/framebuffer_SDL2_0.lua (hunk 1 above)
Comment (Frenzie) [reply ffi/framebuffer_SDL2_0.lua:34]: Yes, we could change one of them if desired for clarity. The construct occurs a fair bit throughout the code dealing with SDL I'm sure. ;-)

The one called `S` in `SDL2_0.lua` is called `SDL` in framebuffer here.
https://github.com/koreader/koreader-base/blob/df96630c21ff9c5ededa1a864da2c048e4c542ed/ffi/SDL2_0.lua#L61-L67

**Comment by Frenzie** [comment]
Afaik it should be perfect on Windows… potentially with some additional hacks to rub Windows' nose in the fact that yes, we can do HiDPI. ;-P

**Pull request merged**